pkg/bitwarden/cli.go

```diff
@@ -0,0 +1,118 @@
+package bitwarden
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+)
+
+type cliClient struct {
+	username string
+	password string
+	sync.Mutex
+	session    string
+	savedItems []Item
+	run        func(cmd string, args ...string) ([]byte, error)
+}
+
+func newCliClient(username, password string) (Client, error) {
+	return newCliClientWithRun(username, password, func(cmd string, args ...string) (bytes []byte, err error) {
+		logrus.WithField("cmd", cmd).WithField("args", args).Info("running command ...")
+		return exec.Command(cmd, args...).Output()
+	})
+}
+
+func newCliClientWithRun(username, password string, run func(cmd string, args ...string) (bytes []byte, err error)) (Client, error) {
+	client := cliClient{
+		username: username,
+		password: password,
+		run:      run,
+	}
+	return &client, client.loginAndListItems()
+}
+
+type bwLoginResponse struct {
+	Success bool `json:"success"`
+	Data    struct {
+		Raw string `json:"raw"`
+	} `json:"data"`
+}
+
+func (c *cliClient) loginAndListItems() error {
+	c.Lock()
+	defer c.Unlock()
+	output, err := c.run("bw", "login", c.username, c.password, "--response")
+	if err != nil {
+		return err
+	}
+	r := bwLoginResponse{}
+	if err := json.Unmarshal(output, &r); err != nil {
+		//TODO need to censor if we plan this on a job
```

Comment:
Should just use the censoring logger straight away

Reply:
Added a commit for protecting `bw-password`.